flake-rule/src/main/java/com/palantir/flake/ShouldRetry.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.flake;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * An annotation signalling that a given test class or test method should be retried in the event of a test failure.
+ * In the event that both a test class and a test method are annotated with this
+ *
+ * Note that this annotation will only actually cause a test to be retried if it is run with the
```

Comment:
I wonder if there's a way around this?  Or if we could add a Findbugs rule that catches cases where you forget to add the rule.  Definitely for the future and not now.

Reply:
Yep, a Baseline rule sounds plausible. If the annotation is present but the rule is not, there's almost certainly something wrong.